### PR TITLE
Type-to-select component dropdown

### LINKED ISSUE #43: Allow typing in the component select
[Original feature request]( https://github.com/zendframework/zf-mkdoc-theme/pull/34#issuecomment-389575846) from @weierophinney in #34:

> Allow typing in the component select (fuzzy match, basically)

### PR DESCRIPTION
**Pull request by weierophinney**
Uses [Choices.js](https://github.com/jshjohnson/Choices) to render a selectable dropdown of components. The component list is fetched via the usual method, parsed into its various optgroups, and finally used to populate the dropdown. The current page is selected by default; selecting any other page redirects to that documentation. Each item gives the package title, and the (shortened) package name.  Users may type to filter the dropdown after first expanding it.

I have amended the styles to omit the "itemSelectText" from the dropdown, so that the title/package text spans the full width of the dropdown.

Fixes #43

### CODE REVIEW

File: theme/component-selector.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="component-selector">
     <h4 class="component-selector__title">Components</h4>
     <h5 class="component-selector__subtitle">Find documentation…</h5>
-    <select class="form-control component-selector__control">
+    <select id="component-selector-control" class="form-control component-selector__control">
```
Comment (froschdesign) [inline theme/component-selector.html]: Is the `id` attribute really needed? I can find no reference.

File: asset/sass/_elements.scss [hunk 2]
```diff
@@ -30,3 +30,18 @@ pre[class*="language-"] {
     margin-top: $padding-large-vertical * 2;
     margin-bottom: $padding-large-vertical * 2;
 }
+
+.dropdown_packagename {
```
Comment (froschdesign) [inline asset/sass/_elements.scss]: This additions should be moved under the `blocks` folder. (a new file like `_choices.scss` or added to the file `_component-selector.scss`)

Comment (froschdesign) [reply asset/sass/_elements.scss]: And the name should follow the block: `choices__packagename

File: asset/js/component-list.js [hunk 3]
```diff
@@ -1,75 +1,90 @@
...
-        element.addEventListener('change', function (event) {
-            // Get value
-            var value = event.target.value;
-            if (value.length === 0) {
-                return;
-            }
+        components.forEach(function (component) {
+            const selected = matchActive.test(component.url);
+            const packageName = component.package.split('/');
+            const label = component.name + '<br/><span class="dropdown_packagename">' + packageName[1] + '</span>';
```
Comment (froschdesign) [inline asset/js/component-list.js]: The same here: `choices__packagename`

File: asset/sass/_elements.scss [hunk 4]
```diff
@@ -30,3 +30,18 @@ pre[class*="language-"] {
...
+  font-size: .8em;
+  color: #666666;
```
Comment (froschdesign) [inline asset/sass/_elements.scss]: I think, the color value can be replaced with the Sass variable `$text-color-secondary`, which is `#777`.

File: asset/sass/_elements.scss [hunk 5]
```diff
@@ -30,3 +30,18 @@ pre[class*="language-"] {
     margin-top: $padding-large-vertical * 2;
     margin-bottom: $padding-large-vertical * 2;
 }
+
+.dropdown_packagename {
+  font-size: .8em;
```
Comment (froschdesign) [inline asset/sass/_elements.scss]: And we have also a Sass variable for the font size: `$font-size-small` 😉 

https://github.com/zendframework/zf-mkdoc-theme/blob/master/asset/sass/_settings.scss#L17

File: theme/component-selector.html (hunk 1 above)
Comment (weierophinney) [reply theme/component-selector.html]: Nope, not needed; I thought I'd omitted it in my most recent changes. I'll revert it.

File: asset/sass/_elements.scss (hunk 2 above)
Comment (weierophinney) [reply asset/sass/_elements.scss]: Done.

File: asset/js/component-list.js (hunk 3 above)
Comment (weierophinney) [reply asset/js/component-list.js]: Done.

File: asset/sass/_elements.scss (hunk 4 above)
Comment (weierophinney) [reply asset/sass/_elements.scss]: Done.

File: asset/sass/_elements.scss (hunk 5 above)
Comment (weierophinney) [reply asset/sass/_elements.scss]: Done.

**Comment by weierophinney** [comment]
@froschdesign Thanks for the feedback! I've updated to incorporate everything you suggested, and am now going to merge and start rolling it out.

**Comment by weierophinney** [comment]
Showing unmerged commits, because I rebased in order to pick up the CHANGELOG.

**Pull request closed without merging**